compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -247,29 +258,47 @@ private PackageDescriptor getPackageDescriptor(TomlTableNode tomlTableNode) {
         String version;
 
         if (tomlTableNode.entries().isEmpty()) {
+            reportDiagnostic(MISSING_PKG_INFO_ERROR,
+                    ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
+                    DiagnosticSeverity.WARNING, tomlTableNode.location());
             return PackageDescriptor.from(defaultOrg(), defaultName(this.projectPath), defaultVersion());
         }
 
         TopLevelNode topLevelPkgNode = tomlTableNode.entries().get(PACKAGE);
         if (topLevelPkgNode == null || topLevelPkgNode.kind() != TomlType.TABLE) {
+            reportDiagnostic(MISSING_PKG_INFO_ERROR,
+                    ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
+                    DiagnosticSeverity.WARNING, tomlTableNode.location());
             return PackageDescriptor.from(defaultOrg(), defaultName(this.projectPath), defaultVersion());
         }
 
         TomlTableNode pkgNode = (TomlTableNode) topLevelPkgNode;
         if (pkgNode.entries().isEmpty()) {
+            reportDiagnostic(MISSING_PKG_INFO_ERROR,
```

Comment:
I think we need to inform the user of the default values inferred from the context with all these error messages.

Reply:
Added via https://github.com/ballerina-platform/ballerina-lang/pull/36174/commits/2e9a0e40f72adcbdce684f1f6b8b6ec354508d4f